### PR TITLE
CloudFormation updates (Queue, SecretsManager, and UserData)

### PR DESCRIPTION
**Pull request by DrazenDodik**
Added a few more things here:

- Allow user to set the .vqueue address when creating a stack using the CF template
- Store the Redis password in the AWS SecretsManager (from where the queue and ValohaiMaster can fetch it from)
- Attach a IAM instance profile to the queue machine
- Install https://github.com/valohai/worker-queue when the machine is setup

### CODE REVIEW

File: main.yml [hunk 1]
```diff
@@ -100,6 +105,7 @@ Resources:
     Type: AWS::CloudFormation::Stack
     Properties:
       Parameters:
+        QueueAddress: !Ref QueueAddress
```
Comment (orasimus) [inline main.yml:108]: I'd alphabetize the list of parameters (putting QueueAdress between KeyPair and Subnet).

File: worker-queue.yml [hunk 2]
```diff
@@ -18,7 +18,22 @@ Parameters:
   WorkersSecurityGroup:
     Type: AWS::EC2::SecurityGroup::Id
 
+  QueueAddress:
```
Comment (orasimus) [inline worker-queue.yml:21]: I'd alphabetize the list of parameters (putting QueueAddress between KeyPair and Subnet).

File: worker-queue.yml [hunk 3]
```diff
@@ -41,9 +56,34 @@ Resources:
...
           Description: for Redis over TLS from workers
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CirdIp: 3.251.38.215/32
+          Description: SSH connection for Valohai maintenance
       Tags:
         - Key: Name
           Value: valohai-sg-queue
+  QueueRole:
```
Comment (orasimus) [inline worker-queue.yml:67]: I'd add an empty line between resources.

File: worker-queue.yml [hunk 4]
```diff
@@ -54,10 +94,19 @@ Resources:
...
+            #!/bin/bash
+            apt-get update -y
+            apt-get install awscli -y
+            export SECRET=`aws secretsmanager get-secret-value --secret-id ValohaiRedisSecret --query SecretString --output text --region ${AWS::Region}`
+            curl https://raw.githubusercontent.com/valohai/worker-queue/main/host/setup.sh | sudo QUEUE_ADDRESS=${QueueAddress} REDIS_PASSWORD=$SECRET bash
+            unset $SECRET
       Tags:
         - Key: Name
           Value: valohai-i-queue
-
```
Comment (orasimus) [inline worker-queue.yml:60]: I'd keep an empty line between resources.

File: workers.yml [hunk 5]
```diff
@@ -43,8 +43,8 @@ Resources:
...
+
```
Comment (orasimus) [inline workers.yml:47]: I'd remove the whitespace from the empty line.

**Review by orasimus — CHANGES_REQUESTED** [review]
Amazing! I think this makes the workflow from the customer perspective smoother.

My only concerns are regarding relying on the service restarting long enough.

1) We could consider adding `StartLimitIntervalSec=0` to the worker queue service (https://unix.stackexchange.com/questions/289629/systemd-restart-always-is-not-honored), to ensure systemd continues trying long enough. Of course we can also just request the customer to restart the EC2 machine if need be.

2) I'm not sure whether each restart pings Let's Encrypt, so I hope they don't get angry at us.

Neither of these is critical imo, just something to keep in mind if we run into them.

I feel bad nitpicking about formatting, my comments/change requests are basically linter stuff 😅

**Comment by orasimus** [comment]
@DrazenDodik oh yeah, check if the README would also warrant an update after these changes!

**Comment by DrazenDodik** [comment]
Added S3 and fixed a few other minor issues

**Pull request merged**